core-default/src/main/java/me/aurium/beetle/defaults/task/sync/TimerSleepBehavior.java

```diff
@@ -0,0 +1,91 @@
+package me.aurium.beetle.defaults.task.sync;
+
+import me.aurium.beetle.api.task.futures.SyncTaskerFuture;
+import me.aurium.beetle.api.task.futures.TaskFuture;
+import me.aurium.beetle.api.task.sync.coordination.BlockingBehavior;
+import me.aurium.beetle.api.task.sync.coordination.SyncConstants;
+import me.aurium.beetle.api.task.sync.SyncPulser;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.locks.LockSupport;
+
+public class TimerSleepBehavior implements BlockingBehavior {
+
+    private final int sleepTimer;
+
+    public TimerSleepBehavior(int sleepTimer) {
+        this.sleepTimer = sleepTimer;
+    }
+
+    public TimerSleepBehavior() {
+        this.sleepTimer = 200;
+    }
+
+    @Override
+    public <T> T waitForJoin(SyncTaskerFuture<T> future, SyncPulser pulser) {
+        while (true) {
+            pulser.pulse();
+
+            T result;
+            if (!(result = future.getNow(SyncConstants.CONSTANT())).equals(SyncConstants.CONSTANT())) {
+                return result;
+            }
+
+            LockSupport.parkNanos(this, sleepTimer);
+        }
+    }
+
+    @Override
+    public <T> T waitForGet(SyncTaskerFuture<T> future, SyncPulser pulser) throws InterruptedException, ExecutionException {
+        while (true) {
+            if (Thread.interrupted()) {
+                throw new InterruptedException();
+            }
+
+            pulser.pulse();
+
+            T result;
+            if (!(result = future.getNow(SyncConstants.CONSTANT())).equals(SyncConstants.CONSTANT())) {
```

Comment:
getNow does not throw ExecutionException if the future is completed exceptionally. It throws CompletionException